System Tests fail due to ClassCastException

Many system tests fail due to a `ClassCastException`. For example, `NSGAIISystemTest::testIntegration`:
```
java.lang.ClassCastException: org.evosuite.testsuite.TestSuiteChromosome cannot be cast to org.evosuite.ga.NSGAChromosome
	at org.evosuite.ga.metaheuristics.NSGAIISystemTest.testIntegration(NSGAIISystemTest.java:331)
```
In this particular example, the relevant lines appear to be the following ones:
https://github.com/VoglSebastian/evosuite/blob/91cad2519b06e68bc3f3a1fcfc02d105e661ac8a/master/src/test/java/org/evosuite/ga/metaheuristics/NSGAIISystemTest.java#L322-L331
Line 323 contains an unchecked cast:
```Java
(GeneticAlgorithm<NSGAChromosome>) getGAFromResult(result);
```
It is assumed that `getGAFromResult` yields a `GeneticAlgorithm<NSGAChromosome>`. But in reality, it should be a `GeneticAlgorithm<TestSuiteChromosome>`. If you change the following lines in that respect, too, the system test at least compiles and runs without mentioned `ClassCastException`. However, it will still fail due to an `AssertionError`:
```
java.lang.AssertionError: 
Expected :0.0
Actual   :0.16666666666666669
at org.evosuite.ga.metaheuristics.NSGAIISystemTest.testIntegration(NSGAIISystemTest.java:333)
```

I suspect that the `ClassCastException`s thrown by many other system tests can be fixed in a similar way. However, the `AssertionError` still requires further investigation…

--- a/master/src/test/java/org/evosuite/ga/metaheuristics/NSGAIISystemTest.java
+++ b/master/src/test/java/org/evosuite/ga/metaheuristics/NSGAIISystemTest.java
@@ -36,6 +36,7 @@ import org.evosuite.ga.operators.ranking.CrowdingDistance;
 import org.evosuite.ga.problems.Problem;
 import org.evosuite.ga.problems.multiobjective.SCH;
 import org.evosuite.ga.problems.singleobjective.Booths;
+import org.evosuite.testsuite.TestSuiteChromosome;
 import org.junit.Assert;
 import org.junit.BeforeClass;
 import org.junit.Test;
@@ -319,13 +320,13 @@ public class NSGAIISystemTest extends SystemTestBase
         Object result = evosuite.parseCommandLine(command);
         Assert.assertNotNull(result);
 
-        GeneticAlgorithm<NSGAChromosome> ga =
-				(GeneticAlgorithm<NSGAChromosome>) getGAFromResult(result);
+        GeneticAlgorithm<TestSuiteChromosome> ga =
+				(GeneticAlgorithm<TestSuiteChromosome>) getGAFromResult(result);
 
-        final FitnessFunction<NSGAChromosome> rho = ga.getFitnessFunctions().get(0);
-        final FitnessFunction<NSGAChromosome> ag = ga.getFitnessFunctions().get(1);
+        final FitnessFunction<TestSuiteChromosome> rho = ga.getFitnessFunctions().get(0);
+        final FitnessFunction<TestSuiteChromosome> ag = ga.getFitnessFunctions().get(1);
 
-        List<NSGAChromosome> population = new ArrayList<>(ga.getBestIndividuals());
+        List<TestSuiteChromosome> population = new ArrayList<>(ga.getBestIndividuals());
         population.sort(new SortByFitness<>(rho, false));
 
         Assert.assertEquals(0.0, population.get(0).getFitness(rho), 0.0);

